### PR TITLE
Write epochs summary into a csv file; Link resumed checkpoint

### PR DESCRIPTION
**Pull request by Nash2325138**
## Why do we need this PR?
* It's hard to view the epochs performance without using Tensorboard. Even with Tensorboard, it's hard to collect all the metric numbers and do a systematical analysis over different experiments.

## Other changes
* Add a link to the resumed checkpoint as a reference.

### CODE REVIEW

File: src/pipeline/base_pipeline.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
 from abc import ABC, abstractmethod
 
 import torch
+import pandas as pd
```
Comment (amjltc295) [inline src/pipeline/base_pipeline.py]: pandas is not in env config

File: src/pipeline/base_pipeline.py [hunk 2]
```diff
@@ -248,5 +250,13 @@ def _print_and_write_log(self, epoch, worker_outputs, write=True):
                 if global_config.verbosity >= 1:
                     logger.info(f'    {str(key):20s}: {value:.4f}')
                 if 'elapsed_time' not in key and write:
+                    value = value.item() if isinstance(value, torch.Tensor) else value
+                    epoch_record[f'{loader_name}_{key}'] = [value]
                     # TODO: See if we can use tree-structured tensorboard logging
                     self.writer.add_scalar(f'{loader_name}_{key}', value)
+
```
Comment (amjltc295) [inline src/pipeline/base_pipeline.py]: We may also want to update the comments for the `_print_and_write_log` function

**Review by amjltc295 — CHANGES_REQUESTED** [review]

File: src/pipeline/base_pipeline.py (hunk 1 above)
Comment (amjltc295) [reply src/pipeline/base_pipeline.py]: (perhaps update the environment.yaml)

File: src/pipeline/base_pipeline.py (hunk 2 above)
Comment (Nash2325138) [reply src/pipeline/base_pipeline.py]: Updated

File: src/pipeline/base_pipeline.py (hunk 1 above)
Comment (Nash2325138) [reply src/pipeline/base_pipeline.py]: Actually the following errors start to pop out when we don't update `environment.yml` for some time.
image

Comment (Nash2325138) [reply src/pipeline/base_pipeline.py]: Maybe a better way is to provide a script like this?
```
conda create -n PytorchGoldenTemplate python=3.7 pytorch=1.3.0 pandas ...
source activate PytorchGoldenTemplate
pip install attrs entrypoints ...
```

Comment (Nash2325138) [reply src/pipeline/base_pipeline.py]: Just like [what I just added](https://github.com/amjltc295/pytorch-golden-template/pull/75/commits/954710df214c380a230f4685eede15c737f1ccd6)

Comment (amjltc295) [reply src/pipeline/base_pipeline.py]: I put it to readme.md and updated the environment.yaml as well

**Review by amjltc295 — APPROVED** [review]
